x-pack/plugins/infra/public/components/asset_details/tabs/profiling/functions.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiEmptyPrompt } from '@elastic/eui';
+import { EmbeddableFunctions } from '@kbn/observability-shared-plugin/public';
+import { useAssetDetailsRenderPropsContext } from '../../hooks/use_asset_details_render_props';
+import { useDatePickerContext } from '../../hooks/use_date_picker';
+import { useProfilingFunctionsData } from '../../hooks/use_profiling_functions_data';
+import { useTabSwitcherContext } from '../../hooks/use_tab_switcher';
+import { ContentTabIds } from '../../types';
+
+export function Functions() {
+  const { asset } = useAssetDetailsRenderPropsContext();
+  const { activeTabId } = useTabSwitcherContext();
+  const { getDateRangeInTimestamp } = useDatePickerContext();
+  const { from, to } = getDateRangeInTimestamp();
+
+  const params = useMemo(
+    () => ({
+      hostname: asset.name,
+      from,
+      to,
+      startIndex: 0,
+      endIndex: 10,
+    }),
+    [asset.name, from, to]
+  );
+
+  const { error, loading, response } = useProfilingFunctionsData({
+    isActive: activeTabId === ContentTabIds.PROFILING,
+    params,
+  });
+
+  if (error !== null) {
+    return <ErrorPrompt />;
+  }
+
+  return (
+    <EmbeddableFunctions
+      data={response ?? undefined}
+      isLoading={loading}
+      rangeFrom={from}
+      rangeTo={to}
+    />
+  );
+}
+
+function ErrorPrompt() {
+  return (
+    <EuiEmptyPrompt
+      color="warning"
+      iconType="warning"
+      titleSize="xs"
+      title={
+        <h2>
+          {i18n.translate('xpack.infra.profiling.functions.loadErrorTitle', {
+            defaultMessage: 'Unable to load the Top Functions',
+          })}
+        </h2>
+      }
+      body={
+        <p>
+          {i18n.translate('xpack.infra.profiling.functions.loadErrorBody', {
+            defaultMessage:
+              'There was an error while trying to load top functions data. Try refreshing the page',
+          })}
+        </p>
+      }
+    />
+  );
+}
```

Comment:
I think you could export this to its own file and reuse in both Flamegraph and Functions tabs?